### PR TITLE
sig-storage: initial config for kubernetes-csi/csi-lib-utils

### PR DESCRIPTION
**Pull request by pohly**
kubernetes-csi/csi-lib-utils is a brand-new stand-alone repo with some
Go utility package (one at the moment, more to come). "make test"
inside it enforces coding style and that unit tests pass, which is
something that we want to ensure before merging any PR.

This is the first config for sig-storage, hence the new OWNERS file.

Fixes: kubernetes-csi/csi-lib-utils#2

### CODE REVIEW

**Comment by pohly** [comment]
/cc @msau42  @saad-ali 

If there's a way to test this config fragment before merging, the please let me know - as it stands, it's basically completely untested.

`make test`  in the repo under test of course passes.

File: config/jobs/kubernetes-sigs/sig-storage/csi/csi-lib-utils-config.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+presubmits:
+  github.com/kubernetes-csi/csi-lib-utils:
+  - name: pull-sig-storage-csi-lib-utils
+    always_run: true
```
Comment (krzyzacy) [inline config/jobs/kubernetes-sigs/sig-storage/csi/csi-lib-utils-config.yaml]: you can flip always_run to false to start with, and you can trigger it on your repo with `/test pull-sig-storage-csi-lib-utils`

**Comment by krzyzacy** [comment]
also [create new jobs with pod-utils](https://github.com/kubernetes/test-infra#create-a-new-job) now :-) it's as easy as https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kind/kind-presubmits.yaml#L2-L12 and you can drop most of the args

File: config/jobs/kubernetes-sigs/sig-storage/csi/csi-lib-utils-config.yaml (hunk 1 above)
Comment (pohly) [reply config/jobs/kubernetes-sigs/sig-storage/csi/csi-lib-utils-config.yaml]: I had considered that, but as we don't have a lot going on in that repo I wasn't worried about breaking anything by making the testing mandatory right away.

@msau42 your call - we can also keep it disabled and do another PR here once we know that it works.

**Comment by pohly** [comment]
@krzyzacy Thanks for the pointer to "Create a new job" - when I asked about prow on Slack the other day, I was only pointed to prow's own README.md and existing examples, so I started from those.

Your example uses the `gcr.io/k8s-testimages/kubekins-e2e:v20181129-33929460d-master` image and it's `./hack/ci/build-all.sh` script. But you were just using that as an example for something that uses podutils and not suggesting that I switch to that image, right?

**Comment by pohly** [comment]
@krzyzacy is the directory structure okay? The org is the "kubernetes-csi" working group inside the "sig-storage" SIG, the repo is "csi-lib-utils" - I wasn't quite sure where to put the config .yaml, so I created a directory for "sig-storage" with a "csi" sub-directory for the CSI specific repos.

File: config/jobs/kubernetes-sigs/sig-storage/OWNERS [hunk 2]
```diff
@@ -0,0 +1,5 @@
+approvers:
```
[the remark this answers is not in the document]
Comment (pohly) [reply config/jobs/kubernetes-sigs/sig-storage/OWNERS]: So no "sig-storage", i.e. `config/jobs/kubernetes-sigs/kubernetes-csi/csi-lib-utils/csi-lib-utils-config.yaml` and `config/jobs/kubernetes-sigs/kubernetes-csi/OWNERS`?

[the remark this answers is not in the document]
Comment (msau42) [reply config/jobs/kubernetes-sigs/sig-storage/OWNERS]: No kubernetes-sigs either (it is an actual GitHub org).

**Comment by pohly** [comment]
@msau42 @krzyzacy updated, please have another look.

**Comment by krzyzacy** [comment]
if this is a new org, did you configure webhooks? Also you need to add an entry, something like https://github.com/kubernetes/test-infra/blob/master/prow/plugins.yaml#L300-L308 (you can choose what plugins you want to enable)

**Comment by pohly** [comment]
Sen Lu <notifications@github.com> writes:
> if this is a new org, did you configure webhooks?

The kubernetes-csi org itself isn't new and has been part of Kubernetes
for a while now, so I think the webhooks are in place. For example, the
PR dashboard shows the PRs in the new repo.

> Also you need to add an entry, something like
> https://github.com/kubernetes/test-infra/blob/master/prow/plugins.yaml#L300-L308
> (you can choose what plugins you want to enable)

Isn't that already covered by the entry for "kubernetes-csi"?

https://github.com/kubernetes/test-infra/blob/f61cf6eab16515ffc32a85489d56847aee9f408a/prow/plugins.yaml#L445-L463

Or is there perhaps some plugin missing? I have no idea what each plugin
does, but looking at the other entries perhaps "trigger" is missing?

In other words, add this?

```
kubernetes-csi/csi-lib-utils:
- trigger

```

**Comment by krzyzacy** [comment]
@pohly sorry, missed that 😂 For some reason I read it as a brand new org, nvm

**Comment by pohly** [comment]
@krzyzacy I've added the "trigger" plugin.

File: config/jobs/kubernetes-csi/csi-lib-utils/csi-lib-utils-config.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  github.com/kubernetes-csi/csi-lib-utils:
+  - name: pull-sig-storage-csi-lib-utils
+    always_run: true
+    decorate: true
+    skip_report: false
+    spec:
+      containers:
+      # This image was chosen over the more often used kubekins-e2e because
+      # it is smaller and has everything we need (basically just a Go environment).
+      - image: gcr.io/k8s-testimages/gcloud-in-go:v20180927-6b4facbe6
```
Comment (krzyzacy) [inline config/jobs/kubernetes-csi/csi-lib-utils/csi-lib-utils-config.yaml]: if you just need go, you can also choose to use something like golang:1.11.2

Comment (pohly) [reply config/jobs/kubernetes-csi/csi-lib-utils/csi-lib-utils-config.yaml]: I was trying to use something that might have been pulled already, i.e. something which is also used by other jobs. Not sure whether that makes sense?

But I agree, now that all of the special logic is handled by the pod-utils, a generic image should work just as well.

Either way, I'd like to get this merged and perhaps it can be tweaked further later on. We have quite a few repos under kubernetes-csi that might benefit from closer integration with Prow, so I expect that we'll come back to this.

Comment (krzyzacy) [reply config/jobs/kubernetes-csi/csi-lib-utils/csi-lib-utils-config.yaml]: You might want to control go versions explicitly - feel free to come back and tune this if you are hitting issues.

**Comment by pohly** [comment]
Now that this is merged, should `/test pull-sig-storage-csi-lib-utils` work right away?

I tried that in https://github.com/kubernetes-csi/csi-lib-utils/pull/4#issuecomment-444275470 but so far nothing has been triggered, at least nothing is visible in github.

File: config/jobs/kubernetes-csi/csi-lib-utils/csi-lib-utils-config.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+presubmits:
+  github.com/kubernetes-csi/csi-lib-utils:
```
Comment (krzyzacy) [inline config/jobs/kubernetes-csi/csi-lib-utils/csi-lib-utils-config.yaml]: ooooops, drop github.com here 😂

Comment (pohly) [reply config/jobs/kubernetes-csi/csi-lib-utils/csi-lib-utils-config.yaml]: Thanks for debugging it. Fix is in https://github.com/kubernetes/test-infra/pull/10340.
